ipywidgets/static/widgets/js/manager.js

```diff
@@ -280,11 +280,26 @@ define([
         /**
          * Handle when a comm is opened.
          */
-        return this.create_model({
-            model_name: msg.content.data.model_name,
-            model_module: msg.content.data.model_module,
-            comm: comm,
-        }).catch(utils.reject("Couldn't create a model.", true));
+        // For < 4.x
+        var model;
+        if (msg.content.data.model_name) {
```

Comment:
> I think that for each registered kernel, we should have the Javascript in a different directory, (one per installed kernelspec). The requirejs path would be automatically prefixed with a string corresponding to the current kernel.

I think this is a good idea and I'm +1 for it - it should be addressed with the other packaging problems https://jupyter.hackpad.com/Packaging-crate-PbIgxnC71or .  @shashi with the  feature @SylvainCorlay is describing, packaging would be more like npm.  You, in the Julia widgets, would depend on a specific widget JS version X.  That would allow us to continue chugging along without affecting you, and when you were ready to upgrade, you could.

@SylvainCorlay despite that, I still think for things like this, which are "widget msg spec" changing, we need deprecation warnings.  Julia (@shashi) and Haskell (@sumitsahrawat) implementations both exist now, and we should do our best to help them transition to our latest Javascript.